### PR TITLE
blockchain shuffle

### PR DESCRIPTION
**Pull request by michaeldjeffrey**
- [x] add chain dead env var
- When chain is dead
  - [x] don't start blockchain related workers when the chain is dead
  - [x] don't route packets through state channels
  - [x] write packet reports to s3
    - [ ] servers have s3 access (ops) @macpie 
  - [x] gateway locations from config service
    - [ ] servers have config service access (ops) @macpie

### CODE REVIEW

File: config/sys.config.src [hunk 1]
```diff
@@ -11,6 +12,17 @@
             {secret, <<"${PP_CONSOLE_SECRET}">>},
             {auto_connect, "${PP_CONSOLE_ENABLE}"}
         ]},
+        {packet_reporter, #{
+            aws_bucket => <<"${PP_PACKET_REPORTER_AWS_BUCKET}">>,
+            aws_bucket_region => <<"${PP_PACKET_REPORTER_AWS_BUCKET_REGION}">>,
+            report_interval => 300_000,
+            report_max_size => 50_000_000
+        }},
+        {ics, #{
```
Comment (macpie) [inline config/sys.config.src:21]: Is this still used if channel is started bellow?

File: src/grpc/pp_ics_gateway_location_worker.erl [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        port = Port,
+        conn_backoff = Backoff
+    }}.
+
+handle_call({get, PubKeyBin}, _From, #state{conn_backoff = Backoff0} = State) ->
+    HotspotName = pp_utils:animal_name(PubKeyBin),
+    case get_gateway_location(PubKeyBin, State) of
+        {error, Reason, true} ->
+            {Delay, Backoff1} = backoff:fail(Backoff0),
+            _ = erlang:send_after(Delay, self(), ?INIT),
```
Comment (macpie) [inline src/grpc/pp_ics_gateway_location_worker.erl:121]: Probably should cleanup the all init thing if we are just using the `.config` to start the channel

File: src/pp_utils.erl [hunk 3]
```diff
@@ -35,21 +48,11 @@ init_ets() ->
...
-        Chain ->
-            Chain
-    end.
-
--spec get_ledger() -> blockchain_ledger_v1:ledger().
-get_ledger() ->
-    blockchain:ledger(get_chain()).
+-spec init_location_cache() -> ok.
+init_location_cache() ->
+    {ok, HLC} = cream:new(200_00, [{initial_capacity, 20_000}, {seconds_to_live, 600}]),
```
Comment (macpie) [inline src/pp_utils.erl:53]:
```suggestion
    {ok, HLC} = cream:new(20_000, [{initial_capacity, 20_000}, {seconds_to_live, 600}]),
```

File: test/pp_test_ics_gateway_service.erl [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+-spec init(atom(), StreamState :: grpcbox_stream:t()) -> grpcbox_stream:t().
+init(_RPC, StreamState) ->
+    StreamState.
+
+-spec handle_info(Msg :: any(), StreamState :: grpcbox_stream:t()) -> grpcbox_stream:t().
+handle_info(_Msg, StreamState) ->
+    StreamState.
+
+region_params(_Ctx, _Msg) ->
+    {grpc_error, {12, <<"UNIMPLEMENTED">>}}.
```
Comment (macpie) [inline test/pp_test_ics_gateway_service.erl:26]:
```suggestion
    {grpc_error, {<<"12">>, <<"UNIMPLEMENTED">>}}.
```

File: test/pp_test_ics_gateway_service.erl [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+load_region(_Ctx, _Msg) ->
+    {grpc_error, {12, <<"UNIMPLEMENTED">>}}.
```
Comment (macpie) [inline test/pp_test_ics_gateway_service.erl:29]:
```suggestion
    {grpc_error, {<<"12">>, <<"UNIMPLEMENTED">>}}.
```

File: src/pp_utils.erl (hunk 3 above)
Comment (michaeldjeffrey) [reply src/pp_utils.erl:53]: I did mean 200k

**Review by macpie — APPROVED** [review]

**Pull request merged**